system/metrics/testing/metrictest.cpp

```diff
@@ -98,7 +111,7 @@ int main(int argc, char *argv[])
     pEventCountMetric = std::make_shared<CounterMetric>("requests", "The number of requests");
     myReporter.addMetric(pEventCountMetric);
 
-    pQueueSizeMetric = std::make_shared<GaugeMetric>("queuesize", "request queue size");
+    pQueueSizeMetric = createMetricAndAddToReporter<GaugeMetric>("queuesize", "request queue size");
     myReporter.addMetric(pQueueSizeMetric);
```

Comment:
I think this should report an error for a duplicate metric.  It would be a good unit test, but I don't think it is intended.

Reply:
The framework should throw an error if a duplicate named metric is added. However, this was not the intent of the change. I was adding a use of the function template that creates and adds a metric. Removed the extra addMetric call.